velox/core/SimpleFunctionMetadata.h

```diff
@@ -394,6 +394,9 @@ class ISimpleFunctionMetadata {
   virtual bool physicalSignatureEquals(
       const ISimpleFunctionMetadata& other) const = 0;
   virtual std::string helpMessage(const std::string& name) const = 0;
+  /// Returns string of all the fields such as logical signature,
+  /// physical signature and priority.
+  virtual std::string toDebugString() const = 0;
```

Comment:
Does this need to be a virtual method?

Reply:
Yes, `using Metadata = core::ISimpleFunctionMetadata`, I add the function to `ISimpleFunctionMetadata` and implement it in `SimpleFunctionMetadata`, otherwise, we will need to convert Metadata to SimpleFunctionMetadata to invoke it.

So I make it virtual function as `helpMessage`.

Reply:
I'm not sure I understand. helpMessage requires different implementation for different functions, but toDebugString seems to be the same for all of them. It seems like toDebugString can be implemented using only public API of ISimpleFunctionMetadata. Is this not the case?

Reply:
Thanks for your explain, now I catch it. I have moved implementation to `ISimpleFunctionMetadata`

Reply:
The compilation failed, the `isVariadic()` function relies on `SimpleFunctionMetadata` template argument `Args...`, so I may need to revert the commit. Do you think so?

Reply:
Can you not use signature().variableArity() ?

Reply:
Thanks! It works well.